### PR TITLE
Feat/earn

### PR DESCRIPTION
**Pull request by jjBlockchain**
## Description (optional)

* Change rewards to earn
* add staking flow

https://www.loom.com/share/2d0baee929f24b5c94469ae4ca9d0331

#### Requirements
* Need to have a verified tier 2 account
* Account needs to be in argentina
* Needs eth on the trading or private key account
#### Steps
* Login
* got to /earn
* Table columns are sortable
* Click on Stake (should be the first item in the row)
* Click continue once the Warning modal pops up
* Select either your trading account or private key account
* Enter below the min amount (1 USD) should pop up with an error
* Enter above the max amount (over your account amount) should also pop up with an error
* Enter amount from 1 USD to your account amount to deposit
* Check the 2 terms
* Click confirm
* This should take you to the manage screen.

### CODE REVIEW

File: packages/blockchain-wallet-v4-frontend/src/modals/Staking/AccountSummary/AccountSummary.detail.template.tsx [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+        {text}
+      </Text>
+      {subText && (
+        <SubTextContainer>
+          <Text color='grey700' size='14px' weight={500}>
+            {subText}
+          </Text>
+          {tooltipId && (
+            <TooltipHost id={tooltipId}>
+              <Icon name='info' size='12px' color='grey400' />
```
Comment (blockdylanb) [inline packages/blockchain-wallet-v4-frontend/src/modals/Staking/AccountSummary/AccountSummary.detail.template.tsx:25]: Not a big deal but we should start using constellation components as much as possible from now on. With the eventual goal of deleting the `blockchain-info-components` repo. Especially for icons like the one you use here: https://blockchain.github.io/blockchain-wallet-v4-frontend/?path=/story/constellation_base-icon--icon&args=icon:IconInformation

File: packages/blockchain-wallet-v4-frontend/src/modals/Staking/AccountSummary/AccountSummary.template.success.tsx [hunk 2]
```diff
@@ -0,0 +1,383 @@
...
+  const currencySymbol = Exchange.getSymbol(walletCurrency) as string
+
+  const accountBalanceBase = account && account.balance
+  const stakingBalanceBase = account && account.totalRewards
+
+  const isDepositEnabled = stakingEligible[coin] ? stakingEligible[coin]?.eligible : false
+  const { commission, rate } = stakingRates[coin]
+
+  return (
+    <Wrapper>
```
Comment (blockdylanb) [inline packages/blockchain-wallet-v4-frontend/src/modals/Staking/AccountSummary/AccountSummary.template.success.tsx:65]: You can get rid of a lot of this flyout structure code by importing FlyoutContainer, FlyoutHeader, FlyoutContent, FlyoutFooter components.

File: packages/blockchain-wallet-v4-frontend/src/modals/Staking/AccountSummary/index.tsx [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+      dispatch(
+        actions.analytics.trackEvent({
+          key: Analytics.WALLET_STAKING_DETAIL_VIEWED,
+          properties: {
+            currency: coin
+          }
+        })
+      )
+    }
+  }, [coin])
```
Comment (blockdylanb) [inline packages/blockchain-wallet-v4-frontend/src/modals/Staking/AccountSummary/index.tsx:39]: Should probably add `isFiatCurrencySupported` to the array here

File: packages/blockchain-wallet-v4-frontend/src/modals/Staking/AccountSummary/index.tsx [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+
+  const handleDepositClick = () => {
```
Comment (blockdylanb) [inline packages/blockchain-wallet-v4-frontend/src/modals/Staking/AccountSummary/index.tsx:41]: You could technically get a little performance boost if you turn these into a hook like `useCallback(() => {...}, [coin])`

File: packages/blockchain-wallet-v4-frontend/src/modals/Staking/AccountSummary/index.tsx [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+      showSupply={showSupply}
+      stepMetadata={stepMetadata}
+      walletCurrency={walletCurrency}
+    />
+  ) : (
+    <Unsupported handleClose={handleClose} walletCurrency={walletCurrency} />
+  )
+}
+
+// const mapDispatchToProps = (dispatch: Dispatch): LinkDispatchPropsType => ({
```
Comment (blockdylanb) [inline packages/blockchain-wallet-v4-frontend/src/modals/Staking/AccountSummary/index.tsx:99]: still need this?

**Review by blockdylanb — APPROVED** [review]

File: packages/blockchain-wallet-v4-frontend/src/modals/Staking/AccountSummary/index.tsx (hunk 5 above)
Comment (jjBlockchain) [reply packages/blockchain-wallet-v4-frontend/src/modals/Staking/AccountSummary/index.tsx:99]: nice catch

File: packages/blockchain-wallet-v4-frontend/src/modals/Staking/AccountSummary/AccountSummary.model.tsx [hunk 6]
```diff
@@ -0,0 +1,125 @@
+import { PaletteColors } from '@blockchain-com/constellation'
+import styled, { css } from 'styled-components'
+
+import { Text } from 'blockchain-info-components'
+import { FlyoutWrapper } from 'components/Flyout'
+
+const LeftRightContainerCss = css`
```
[the remark this answers is not in the document]
Comment (jjBlockchain) [reply packages/blockchain-wallet-v4-frontend/src/modals/Staking/AccountSummary/AccountSummary.model.tsx:7]: 🔥

**Review by blockdylanb — APPROVED** [review]

**Pull request merged**